change-camera-controller/src/main/java/com/esri/arcgismaps/sample/changecameracontroller/MainActivity.kt

```diff
@@ -0,0 +1,252 @@
+        graphicsOverlay.graphics.add(airplane3DGraphic)
+
+        sceneView.apply {
+            // set the scene to the SceneView
+            this.scene = scene
+            // add the graphics overlay to the SceneView
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        lifecycleScope.launch {
+            // if the map load failed show an error and return
+            scene.load().onFailure {
+                showError("Failed to load the scene: ${it.message}")
+                return@launch
+            }
+            // set the sceneView viewpoint to the default camera
+            sceneView.setViewpointCamera(defaultCamera)
+            // copy assets to the cache directory if needed
+            copyAssetsToCache(assetFiles, cacheDir, false)
+            // load the airplane model file and update the the airplane3DGraphic
+            loadModel(getString(R.string.bristol_model_file), airplane3DGraphic)
+        }
+    }
+
+    /**
+     * Copies the list of [assets] files from the assets folder to a given [cache] directory. This
+     * suspending function runs on the [Dispatchers.IO] context. If [overwrite] is true, any assets
+     * already in the [cache] directory are overwritten, otherwise copy is skipped
+     */
+    private suspend fun copyAssetsToCache(
+        assets: List<String>,
+        cache: File,
+        overwrite: Boolean
+    ) = withContext(Dispatchers.IO) {
+        // get the AssetManager
+        val assetManager = applicationContext.assets ?: return@withContext
+        assets.forEach { assetName ->
+            // check for cancellation before reading/writing the asset files
+            ensureActive()
+            try {
+                // create an output file in the cache directory
+                val outFile = File(cache, assetName)
+                // if the output file doesn't exist or overwrite is enabled
+                if (!outFile.exists() || overwrite) {
+                    // create an input stream to the asset
+                    assetManager.open(assetName).use { inputStream ->
+                        // create an file output stream to the output file
+                        FileOutputStream(outFile).use { outputStream ->
+                            // copy the input file stream to the output file stream
+                            inputStream.copyTo(outputStream)
+                        }
+                        Log.i(TAG, "$assetName copied to cache.")
+                    }
+                } else {
+                    Log.i(TAG, "$assetName already in cache, skipping copy.")
+                }
+            } catch (exception: Exception) {
+                showError("Error caching asset :${exception.message}")
+            }
+        }
+    }
+
+    /**
+     * Loads a [ModelSceneSymbol] from the [filename] in the [getCacheDir] and updates the [graphic]
+     */
+    private suspend fun loadModel(filename: String, graphic: Graphic) {
```

Comment:
Could you move this function to below the `onCreate` as this function showcases the use of creating a `ModelSceneSymbol`. We could move `copyAssetsToCache` to be below `loadModel` as it just is a function for configuration purposes.